src/org/traccar/protocol/ArnaviBinaryProtocolDecoder.java

```diff
@@ -0,0 +1,273 @@
+            reply.writeByte((byte) 0x7B);
+            reply.writeByte((byte) 0x04);
+            reply.writeByte((byte) 0x00);
+            reply.writeByte((byte) crc8(timestampBytes));
+            reply.writeBytes(timestampBytes);
+            reply.writeByte((byte) 0x7D);
+        }
+        if (channel != null) {
+            channel.write(reply);
+        }
+    }
+
+    private void sendConfirmationPackage(Channel channel, int packageCounter) {
+        if (channel == null) {
+            return;
+        }
+        ChannelBuffer reply = ChannelBuffers.directBuffer(ByteOrder.LITTLE_ENDIAN, 4);
+        byte[] replBytes = new byte[]{0x7B, 0x00, (byte) packageCounter, 0x7D};
+        reply.writeBytes(replBytes);
+        channel.write(reply);
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        ChannelBuffer buf = (ChannelBuffer) msg;
+        int length = buf.readableBytes();
+
+        if (length == 10) {
+            short signature = buf.readUnsignedByte(); // header
+            if (signature != 0xff) {
+                return null;
+            }
+
+            short protocolVersion = buf.readUnsignedByte(); // protocol version. Must be 0x22 or 0x23
+            if (protocolVersion != PROTOCOL_VERSION1 && protocolVersion != PROTOCOL_VERSION2) {
+                return null;
+            }
+
+            String imei = String.valueOf(buf.readLong());
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, imei);
+            sendConfirmationHeader(channel, protocolVersion);
+
+        } else {
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress);
+            if (deviceSession != null) {
+                positions = new LinkedList<>();
+                short signature = buf.readUnsignedByte(); // header of package
+
+                if (signature == 0x5B) {
+                    // standart package signature
+                    byte[] currentPackage = new byte[length - 1];
+                    buf.readBytes(currentPackage);
+                    parsePackage(currentPackage, channel, deviceSession);
+                }
+
+                return positions;
+            } else {
+                System.out.println("deviceSession is null while PACKAGE got!");
+            }
+        }
+        System.out.println("decode return null\n");
+        return null;
+    }
+
+    private void parsePackage(byte[] packageRawData, Channel channel, DeviceSession deviceSession) {
+        int packageCounter = 0xFE;                                       // package has not been tested by default
+        int packageLength = packageRawData.length;
+        ChannelBuffer packageBuffer = ChannelBuffers.directBuffer(ByteOrder.LITTLE_ENDIAN, packageLength);
+        packageBuffer.writeBytes(packageRawData);
+
+        packageCounter = packageBuffer.readUnsignedByte();
```

Comment:
This is not `packageCounter`, It's parcel number which just showing a sending sequence number.